Use "pkg-config gnutls" instead of "libgnutls-config" which was removed
in gnutls 2.8.0.

Use "gnutls_priority_set_direct" instead of
"gnutls_certificate_type_set_priority" which was removed in gnutls 3.4.0.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=624049

Don't use "CTYPE-OPENPGP" because OpenPGP support was removed in gnutls
3.6.0.

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=879531
--- configure.orig	2007-04-09 14:53:04.000000000 -0500
+++ configure	2010-04-01 03:29:39.000000000 -0500
@@ -20144,8 +20144,8 @@
 _ACEOF
 
    GNUTLS=1
-   CPPFLAGS="${CPPFLAGS} `libgnutls-config --cflags`"
-   LDFLAGS="${LDFLAGS} `libgnutls-config --libs`"
+   CPPFLAGS="${CPPFLAGS} `pkg-config gnutls --cflags`"
+   LDFLAGS="${LDFLAGS} `pkg-config gnutls --libs`"
    if test "$withval" != "yes"; then
 	GNUTLSROOT=$withval
 	LDFLAGS="${LDFLAGS} -L$GNUTLSROOT/lib"
@@ -22926,7 +22926,7 @@
 { echo "$as_me:$LINENO: result: $ac_cv_lib_gnutls_gnutls_global_init" >&5
 echo "${ECHO_T}$ac_cv_lib_gnutls_gnutls_global_init" >&6; }
 if test $ac_cv_lib_gnutls_gnutls_global_init = yes; then
-  LIBS="${LIBS} `libgnutls-config --libs`"
+  LIBS="${LIBS} `pkg-config gnutls --libs`"
 else
   { { echo "$as_me:$LINENO: error: Get the GNU TLS library (http://www.gnutls.org/)" >&5
 echo "$as_me: error: Get the GNU TLS library (http://www.gnutls.org/)" >&2;}
--- echoping.c
+++ echoping.c
@@ -137,9 +137,6 @@ main(argc, argv)
 	gnutls_session  session;
 	gnutls_certificate_credentials xcred;
 	int             tls_result;
-	const int       cert_type_priority[3] = { GNUTLS_CRT_X509,
-		GNUTLS_CRT_OPENPGP, 0
-	};
 #endif
 
 	int             priority;
@@ -1020,9 +1017,9 @@ main(argc, argv)
 					if (tls_result != 0)
 						err_sys
 						    ("Cannot create a new TLS session");
-					gnutls_set_default_priority(session);
-					gnutls_certificate_type_set_priority(session,
-									     cert_type_priority);
+					gnutls_priority_set_direct(session,
+							"NORMAL",
+							NULL);
 					gnutls_credentials_set(session,
 							       GNUTLS_CRD_CERTIFICATE,
 							       xcred);
